### ISSUE TITLE
We need a system for adding documentation to feature PRs

### ISSUE THREAD

**Issue by helderco** [dagger/dagger#2390]
After #1510 I started including docs in feature PRs:

- https://github.com/dagger/dagger/issues/1510#issuecomment-1026133473

The problem is that you'll have new feature documentation published after merge, but the feature itself won't be available until the next release, unless you build from main locally.

We can just not include the new page in the sidebar, but sometimes we want to update an existing page. Example: 
- https://github.com/dagger/dagger/pull/2265

One possible solution is to just add an admonition like _"New in version 0.2.10"_, but in the case of updating an existing doc page, it would remove actually current information before the release.

Is there a smarter way to solve this?

/cc @slumbering

**Comment by slumbering** [comment]
sorry @helderco, I've missed this issue.

Does the option **draft** in Markdown front matter could help ?
`A boolean flag to indicate that a document is a work-in-progress. Draft documents will only be displayed during development.`
https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#draft

**Comment by helderco** [comment]
Oh, that's better than just not including the new page in the sidebar which is a current workaround, but the bigger problem is when we want to update an already existing page in a PR. This happened in:

- https://github.com/dagger/dagger/pull/2265

**Comment by helderco** [comment]
A lot has changed. What's our current process on this @vikram-dagger?

**Comment by vikram-dagger** [comment]
I would say there are two approaches:

#1
- add/update the page as part of your PR 
- merge it when your PR is approved (this will not publish it)
- manually publish it when ready via Netlify

The problem here is that other docs changes cannot be published, as doing so will also publish your change. So then you have:

#2
- add/update your page as a separate PR (connect with your main PR via comments or sth)
- wait for approval but do not merge until ready to publish
- merge and manually publish via Netlify

I'm not sure if there are other approaches?
